src/repository.js

```diff
@@ -105,7 +105,7 @@ export class Repository {
     }
 
     return findQuery
-      .then(x => this.populateEntities(x))
+      .then(x => x ? this.populateEntities(x) : Promise.reject(new Error('No data')))
```

Comment:
what do you prefer
`.then(x= > x ? this.populateEntities(x) : Promise.reject(new Error('No data')))`
or
```js
.then(x= > {
  if (!x) {
   throw new Error('No data');
  }
  return this.populateEntities(x);
}
```
i prefer the former